### DISCUSSION TITLE
make_a_offset.csh

### DISCUSSION THREAD

**Discussion by KIKINEO** [gmtsar/gmtsar#514] [Q&A] [answered]
@Xiaohua-Eric-Xu Dear Dr. Xu, recently I am using your script file make_a_offset.csh to process the Sentinel 1 data to get the image azimuth shape variables, I uploaded the log file for the whole process of processing the data, according to the file description, after pre-processing the image data I ran this script file in F1, F2, F3 respectively, but when the program started to run there were some errors. I don't know how to deal with it, whether it's a problem with the wrong path setting of the file or a problem with the input of the parameters before I start, I am not sure now. variables, how do I combine the final results on a single plot?

root@ubuntu:~/M-SAR/F1# make_a_offset.csh S1_20220105_111101_F1.PRM S1_20220117_111101_F1.PRM 8 2 8 2 1
make_a_offset.csh S1_20220105_111101_F1.PRM S1_20220117_111101_F1.PRM 8 2 8 2
mkdir: cannot create directory ‘azi_offset’: File exists
ln: failed to create symbolic link './S1_20220105_111101_F1.SLC': File exists
ln: failed to create symbolic link './S1_20220117_111101_F1.SLC': File exists
ln: failed to create symbolic link './trans.dat': File exists
ln: failed to create symbolic link './dem.grd': File exists
ground velocity:  6795.42
azi pixel size, 13.968
 setting nx to 8
 setting ny to 2
 setting xsearch to 8
 setting nx_corr to 16
 setting ysearch to 2
 setting ny_corr to 4
 ignoring shift in prm
 setting ashift and rshift to zero
 setting ashift and rshift to zero
 21580 12188 21580 12188 0 0 0.000000
 locations  n 16 nx 21580 nyl 2 nxl 8 x_inc 1957 y_inc 4058
 elapsed time: 0.129500 
xinc:20124  yinc:48696
blockmedian [WARNING]: (x_max-x_min) must equal (NX + eps) * x_inc), where NX is an integer and |eps| <= 0.0001.
blockmedian [WARNING]: (y_max-y_min) must equal (NY + eps) * y_inc), where NY is an integer and |eps| <= 0.0001.
blockmedian (gmtapi_init_grdheader): Please select compatible -R and -I values
xyz2grd [WARNING]: (x_max-x_min) must equal (NX + eps) * x_inc), where NX is an integer and |eps| <= 0.0001.
xyz2grd [WARNING]: (y_max-y_min) must equal (NY + eps) * y_inc), where NY is an integer and |eps| <= 0.0001.
xyz2grd (gmtapi_init_grdheader): Please select compatible -R and -I values
surface [INFORMATION]: Provides 3, expects 3-column binary data
surface [INFORMATION]: Cartesian input grid
surface [INFORMATION]: Cartesian input grid
surface [INFORMATION]: Cannot suggest any n_columns,n_rows better than your current -R -I settings.
surface [INFORMATION]: Cartesian input grid
surface [INFORMATION]: Cartesian input grid
surface [INFORMATION]: Cartesian input grid
surface [INFORMATION]: Cartesian input grid
surface [ERROR]: Grid must have at least 4 nodes in each direction (you have 737 by 2) - abort.
surface [INFORMATION]: Provides 3, expects 3-column binary data
surface [INFORMATION]: Cartesian input grid
surface [INFORMATION]: Cartesian input grid
surface [INFORMATION]: Cannot suggest any n_columns,n_rows better than your current -R -I settings.
surface [INFORMATION]: Cartesian input grid
surface [INFORMATION]: Cartesian input grid
surface [INFORMATION]: Cartesian input grid
surface [INFORMATION]: Cartesian input grid
surface [ERROR]: Grid must have at least 4 nodes in each direction (you have 737 by 2) - abort.
grdtrack [ERROR]: Cannot find file raln.grd
grdtrack [ERROR]: Cannot find file ralt.grd
grdtrack [ERROR]: Must specify -G at least once
rm: No match.
gmtinfo [ERROR]: No input data found!
gmtinfo [ERROR]: No input data found!
gmtinfo [ERROR]: No input data found!
gmtinfo [ERROR]: No input data found!
xinc2:-96  yinc2:-24
xyz2grd [ERROR]: Option -R parsing failure. Correct syntax:

  -R<west>/<east>/<south>/<north>[+r]
     Specify the min/max coordinates of your data region in user units. Use
     dd:mm[:ss] for regions given in arc degrees, minutes [and seconds]. Use
     -R<xmin>/<xmax>/<ymin>/<ymax>[+u<unit>] for regions given in projected
     coordinates, with <unit> selected from e|f|k|M|n|u [e]. If +u is set,
     projected regions centered on (0,0) may beset via -R<halfwidth>[/⏎
     …<halfheight>]+u<unit>, where <halfheight> defaults to <halfwidth> if not
     given. Use [yyyy[-mm[-dd]]]T[hh[:mm[:ss[.xxx]]]] format for time axes.
     Append +r if -R specifies the coordinates of the lower left and upper
     right corners of a rectangular area.
     Alternatively, use -R<code><x0>/<y0>/<n_columns>/<n_rows> for origin and
     grid dimensions, where <code> is a 2-char combo from [T|M|B][L|C|R] (top/⏎
     …middle/bottom/left/center/right) and grid spacing must be specified via
     -I<dx>[/<dy>] (also see -r).
xyz2grd [ERROR]: Offending option -R///
grdmath [ERROR]: aoff_ll.grd is not a number, operator or file name
grdsample [ERROR]: Option -R parsing failure. Correct syntax:

  -R<west>/<east>/<south>/<north>[+r]
     Specify the min/max coordinates of your data region in user units. Use
     dd:mm[:ss] for regions given in arc degrees, minutes [and seconds]. Use
     -R<xmin>/<xmax>/<ymin>/<ymax>[+u<unit>] for regions given in projected
     coordinates, with <unit> selected from e|f|k|M|n|u [e]. If +u is set,
     projected regions centered on (0,0) may beset via -R<halfwidth>[/⏎
     …<halfheight>]+u<unit>, where <halfheight> defaults to <halfwidth> if not
     given. Use [yyyy[-mm[-dd]]]T[hh[:mm[:ss[.xxx]]]] format for time axes.
     Append +r if -R specifies the coordinates of the lower left and upper
     right corners of a rectangular area.
     Alternatively, use -R<code><x0>/<y0>/<n_columns>/<n_rows> for origin and
     grid dimensions, where <code> is a 2-char combo from [T|M|B][L|C|R] (top/⏎
     …middle/bottom/left/center/right) and grid spacing must be specified via
     -I<dx>[/<dy>] (also see -r).
grdsample [ERROR]: Offending option -R///
grdgradient [ERROR]: Cannot find file s_dem.grd
grdinfo [ERROR]: Cannot find file azi_offset_ll.grd
grdinfo [ERROR]: Must specify one or more input files
grdinfo [ERROR]: Cannot find file azi_offset_ll.grd
grdinfo [ERROR]: Must specify one or more input files
grdinfo [ERROR]: Cannot find file azi_offset_ll.grd
grdinfo [ERROR]: Must specify one or more input files
grdinfo [ERROR]: Cannot find file azi_offset_ll.grd
grdinfo [ERROR]: Must specify one or more input files
awk: cmd. line:1: (FILENAME=- FNR=1) fatal: division by zero attempted
awk: cmd. line:1: (FILENAME=- FNR=1) fatal: division by zero attempted
grdinfo [ERROR]: Cannot find file azi_offset_ll.grd
grdinfo [ERROR]: Must specify one or more input files
psbasemap [ERROR]: Option -J parsing failure. Correct syntax:

  -J Select map proJection (<scale> in cm/degree, <width> in cm). Modifiers:
       +dh Specifying map height.
       +du Specifying maximum map dimension.
       +dl Specifying minimum map dimension.
       +dw Specifying map width [Default].
     Azimuthal projections set -Rg unless polar aspect or -R<...>+r is set.
     -Ja|A<lon0>/<lat0>[/<hor>]/<scl> (or <radius>/<lat>)|<width> (Lambert
       Azimuthal Equal-area)
     -Jb|B<lon0>/<lat0>/<lat1>/<lat2>/<scl>|<width> (Albers Conic Equal-
       area)
     -Jcyl_stere|Cyl_stere/[<lon0>/[<lat0>/]]<lat1>/<lat2>/<scl>|<width>
       (Cylindrical Stereographic)
     -Jc|C<lon0>/<lat0><scl>|<width> (Cassini)
     -Jd|D<lon0>/<lat0>/<lat1>/<lat2>/<scl>|<width> (Equidistant Conic)
     -Je|E<lon0>/<lat0>[/<horizon>]/<scl> (or <radius>/<lat>)|<width>
       (Azimuthal Equidistant)
     -Jf|F<lon0>/<lat0>[/<horizon>]/<scl> (or <radius>/<lat>)|<width>
       (Gnomonic)
     -Jg|G<lon0>/<lat0>/<scl> (or <radius>/<lat>)|<width> (Orthographic)
     -Jg|G<lon0>/<lat0>/<scl>|<width>[+a<azimuth>][+t<tilt>][+v<vwidth>/⏎
       …<vheight>][+w<twist>][+z<altitude>[r|R]|g] (General Perspective)
     -Jh|H[<lon0>/]<scl>|<width> (Hammer-Aitoff)
     -Ji|I[<lon0>/]<scl>|<width> (Sinusoidal)
     -Jj|J[<lon0>/]<scl>|<width> (Miller)
     -Jkf|Kf[<lon0>/]<scl>|<width> (Eckert IV)
     -Jks|Ks[<lon0>/]<scl>|<width> (Eckert VI)
     -Jl|L<lon0>/<lat0>/<lat1>/<lat2>/<scl>|<width> (Lambert Conformal Conic)
     -Jm|M[<lon0>/[<lat0>/]]<scl>|<width> (Mercator)
     -Jn|N[<lon0>/]<scl>|<width> (Robinson projection)
     -Jo|O (Oblique Mercator). Specify one of three definitions:
       -Jo|O[a|A]<lon0>/<lat0>/<azimuth>/<scl>|<width>[+v]
       -Jo|O[b|B]<lon0>/<lat0>/<lon1>/<lat1>/<scl>|<width>[+v]
       -Jo|Oc|C<lon0>/<lat0>/<lonp>/<latp>/<scl>|<width>[+v]
     -Jpoly|Poly/[<lon0>/[<lat0>/]]<scl>|<width> ((American) Polyconic)
     -Jq|Q[<lon0>/[<lat0>/]]<scl>|<width> (Equidistant Cylindrical)
     -Jr|R[<lon0>/]<scl>|<width> (Winkel Tripel)
     -Js|S<lon0>/<lat0>/[<horizon>/]<scl> (or <slat>/<scl> or <radius>/<lat>)|⏎
       …<width> (Stereographic)
     -Jt|T<lon0>/[<lat0>/]<scl>|<width> (Transverse Mercator)
     -Ju|U[<zone>/]<scl>|<width> (UTM)
     -Jv|V<lon0>/<scl>|<width> (van der Grinten)
     -Jw|W<lon0>/<scl>|<width> (Mollweide)
     -Jy|Y[<lon0>/[<lat0>/]]<scl>|<width> (Cylindrical Equal-area)
     -Jp|P<scl>|<width>[+a][+f[e|p|<radius>]][+k<kind>][+r<offset>][+t<origin]⏎
       …[+z[p|<radius>]] (Polar [azimuth] (theta,radius)).
     -Jx|X<x-scl>|<width>[d|l|p<power>|t|T][/<y-scl>|<height>[d|l|p<power>|t|⏎
       …T]] (Linear, log, and power projections)
     (See basemap for more details on projection syntax)
psbasemap [ERROR]: Offending option -Jmc
grdimage [ERROR]: Option -J parsing failure. Correct syntax:

  -J Select map proJection (<scale> in cm/degree, <width> in cm). Modifiers:
       +dh Specifying map height.
       +du Specifying maximum map dimension.
       +dl Specifying minimum map dimension.
       +dw Specifying map width [Default].
     Azimuthal projections set -Rg unless polar aspect or -R<...>+r is set.
     -Ja|A<lon0>/<lat0>[/<hor>]/<scl> (or <radius>/<lat>)|<width> (Lambert
       Azimuthal Equal-area)
     -Jb|B<lon0>/<lat0>/<lat1>/<lat2>/<scl>|<width> (Albers Conic Equal-
       area)
     -Jcyl_stere|Cyl_stere/[<lon0>/[<lat0>/]]<lat1>/<lat2>/<scl>|<width>
       (Cylindrical Stereographic)
     -Jc|C<lon0>/<lat0><scl>|<width> (Cassini)
     -Jd|D<lon0>/<lat0>/<lat1>/<lat2>/<scl>|<width> (Equidistant Conic)
     -Je|E<lon0>/<lat0>[/<horizon>]/<scl> (or <radius>/<lat>)|<width>
       (Azimuthal Equidistant)
     -Jf|F<lon0>/<lat0>[/<horizon>]/<scl> (or <radius>/<lat>)|<width>
       (Gnomonic)
     -Jg|G<lon0>/<lat0>/<scl> (or <radius>/<lat>)|<width> (Orthographic)
     -Jg|G<lon0>/<lat0>/<scl>|<width>[+a<azimuth>][+t<tilt>][+v<vwidth>/⏎
       …<vheight>][+w<twist>][+z<altitude>[r|R]|g] (General Perspective)
     -Jh|H[<lon0>/]<scl>|<width> (Hammer-Aitoff)
     -Ji|I[<lon0>/]<scl>|<width> (Sinusoidal)
     -Jj|J[<lon0>/]<scl>|<width> (Miller)
     -Jkf|Kf[<lon0>/]<scl>|<width> (Eckert IV)
     -Jks|Ks[<lon0>/]<scl>|<width> (Eckert VI)
     -Jl|L<lon0>/<lat0>/<lat1>/<lat2>/<scl>|<width> (Lambert Conformal Conic)
     -Jm|M[<lon0>/[<lat0>/]]<scl>|<width> (Mercator)
     -Jn|N[<lon0>/]<scl>|<width> (Robinson projection)
     -Jo|O (Oblique Mercator). Specify one of three definitions:
       -Jo|O[a|A]<lon0>/<lat0>/<azimuth>/<scl>|<width>[+v]
       -Jo|O[b|B]<lon0>/<lat0>/<lon1>/<lat1>/<scl>|<width>[+v]
       -Jo|Oc|C<lon0>/<lat0>/<lonp>/<latp>/<scl>|<width>[+v]
     -Jpoly|Poly/[<lon0>/[<lat0>/]]<scl>|<width> ((American) Polyconic)
     -Jq|Q[<lon0>/[<lat0>/]]<scl>|<width> (Equidistant Cylindrical)
     -Jr|R[<lon0>/]<scl>|<width> (Winkel Tripel)
     -Js|S<lon0>/<lat0>/[<horizon>/]<scl> (or <slat>/<scl> or <radius>/<lat>)|⏎
       …<width> (Stereographic)
     -Jt|T<lon0>/[<lat0>/]<scl>|<width> (Transverse Mercator)
     -Ju|U[<zone>/]<scl>|<width> (UTM)
     -Jv|V<lon0>/<scl>|<width> (van der Grinten)
     -Jw|W<lon0>/<scl>|<width> (Mollweide)
     -Jy|Y[<lon0>/[<lat0>/]]<scl>|<width> (Cylindrical Equal-area)
     -Jp|P<scl>|<width>[+a][+f[e|p|<radius>]][+k<kind>][+r<offset>][+t<origin]⏎
       …[+z[p|<radius>]] (Polar [azimuth] (theta,radius)).
     -Jx|X<x-scl>|<width>[d|l|p<power>|t|T][/<y-scl>|<height>[d|l|p<power>|t|⏎
       …T]] (Linear, log, and power projections)
     (See basemap for more details on projection syntax)
grdimage [ERROR]: Offending option -Jmc
grdimage [ERROR]: Option -R parsing failure. Correct syntax:

  -R<west>/<east>/<south>/<north>[+r]
     Specify the min/max coordinates of your data region in user units. Use
     dd:mm[:ss] for regions given in arc degrees, minutes [and seconds]. Use
     -R<xmin>/<xmax>/<ymin>/<ymax>[+u<unit>] for regions given in projected
     coordinates, with <unit> selected from e|f|k|M|n|u [e]. If +u is set,
     projected regions centered on (0,0) may beset via -R<halfwidth>[/⏎
     …<halfheight>]+u<unit>, where <halfheight> defaults to <halfwidth> if not
     given. Use [yyyy[-mm[-dd]]]T[hh[:mm[:ss[.xxx]]]] format for time axes.
     Append +r if -R specifies the coordinates of the lower left and upper
     right corners of a rectangular area.
     Alternatively, use -R<code><x0>/<y0>/<n_columns>/<n_rows> for origin and
     grid dimensions, where <code> is a 2-char combo from [T|M|B][L|C|R] (top/⏎
     …middle/bottom/left/center/right) and grid spacing must be specified via
     -I<dx>[/<dy>] (also see -r).
grdimage [ERROR]: Offending option -R///
grdimage [ERROR]: Option -J parsing failure. Correct syntax:

  -J Select map proJection (<scale> in cm/degree, <width> in cm). Modifiers:
       +dh Specifying map height.
       +du Specifying maximum map dimension.
       +dl Specifying minimum map dimension.
       +dw Specifying map width [Default].
     Azimuthal projections set -Rg unless polar aspect or -R<...>+r is set.
     -Ja|A<lon0>/<lat0>[/<hor>]/<scl> (or <radius>/<lat>)|<width> (Lambert
       Azimuthal Equal-area)
     -Jb|B<lon0>/<lat0>/<lat1>/<lat2>/<scl>|<width> (Albers Conic Equal-
       area)
     -Jcyl_stere|Cyl_stere/[<lon0>/[<lat0>/]]<lat1>/<lat2>/<scl>|<width>
       (Cylindrical Stereographic)
     -Jc|C<lon0>/<lat0><scl>|<width> (Cassini)
     -Jd|D<lon0>/<lat0>/<lat1>/<lat2>/<scl>|<width> (Equidistant Conic)
     -Je|E<lon0>/<lat0>[/<horizon>]/<scl> (or <radius>/<lat>)|<width>
       (Azimuthal Equidistant)
     -Jf|F<lon0>/<lat0>[/<horizon>]/<scl> (or <radius>/<lat>)|<width>
       (Gnomonic)
     -Jg|G<lon0>/<lat0>/<scl> (or <radius>/<lat>)|<width> (Orthographic)
     -Jg|G<lon0>/<lat0>/<scl>|<width>[+a<azimuth>][+t<tilt>][+v<vwidth>/⏎
       …<vheight>][+w<twist>][+z<altitude>[r|R]|g] (General Perspective)
     -Jh|H[<lon0>/]<scl>|<width> (Hammer-Aitoff)
     -Ji|I[<lon0>/]<scl>|<width> (Sinusoidal)
     -Jj|J[<lon0>/]<scl>|<width> (Miller)
     -Jkf|Kf[<lon0>/]<scl>|<width> (Eckert IV)
     -Jks|Ks[<lon0>/]<scl>|<width> (Eckert VI)
     -Jl|L<lon0>/<lat0>/<lat1>/<lat2>/<scl>|<width> (Lambert Conformal Conic)
     -Jm|M[<lon0>/[<lat0>/]]<scl>|<width> (Mercator)
     -Jn|N[<lon0>/]<scl>|<width> (Robinson projection)
     -Jo|O (Oblique Mercator). Specify one of three definitions:
       -Jo|O[a|A]<lon0>/<lat0>/<azimuth>/<scl>|<width>[+v]
       -Jo|O[b|B]<lon0>/<lat0>/<lon1>/<lat1>/<scl>|<width>[+v]
       -Jo|Oc|C<lon0>/<lat0>/<lonp>/<latp>/<scl>|<width>[+v]
     -Jpoly|Poly/[<lon0>/[<lat0>/]]<scl>|<width> ((American) Polyconic)
     -Jq|Q[<lon0>/[<lat0>/]]<scl>|<width> (Equidistant Cylindrical)
     -Jr|R[<lon0>/]<scl>|<width> (Winkel Tripel)
     -Js|S<lon0>/<lat0>/[<horizon>/]<scl> (or <slat>/<scl> or <radius>/<lat>)|⏎
       …<width> (Stereographic)
     -Jt|T<lon0>/[<lat0>/]<scl>|<width> (Transverse Mercator)
     -Ju|U[<zone>/]<scl>|<width> (UTM)
     -Jv|V<lon0>/<scl>|<width> (van der Grinten)
     -Jw|W<lon0>/<scl>|<width> (Mollweide)
     -Jy|Y[<lon0>/[<lat0>/]]<scl>|<width> (Cylindrical Equal-area)
     -Jp|P<scl>|<width>[+a][+f[e|p|<radius>]][+k<kind>][+r<offset>][+t<origin]⏎
       …[+z[p|<radius>]] (Polar [azimuth] (theta,radius)).
     -Jx|X<x-scl>|<width>[d|l|p<power>|t|T][/<y-scl>|<height>[d|l|p<power>|t|⏎
       …T]] (Linear, log, and power projections)
     (See basemap for more details on projection syntax)
grdimage [ERROR]: Offending option -Jmc
grdimage [ERROR]: Option -R parsing failure. Correct syntax:

  -R<west>/<east>/<south>/<north>[+r]
     Specify the min/max coordinates of your data region in user units. Use
     dd:mm[:ss] for regions given in arc degrees, minutes [and seconds]. Use
     -R<xmin>/<xmax>/<ymin>/<ymax>[+u<unit>] for regions given in projected
     coordinates, with <unit> selected from e|f|k|M|n|u [e]. If +u is set,
     projected regions centered on (0,0) may beset via -R<halfwidth>[/⏎
     …<halfheight>]+u<unit>, where <halfheight> defaults to <halfwidth> if not
     given. Use [yyyy[-mm[-dd]]]T[hh[:mm[:ss[.xxx]]]] format for time axes.
     Append +r if -R specifies the coordinates of the lower left and upper
     right corners of a rectangular area.
     Alternatively, use -R<code><x0>/<y0>/<n_columns>/<n_rows> for origin and
     grid dimensions, where <code> is a 2-char combo from [T|M|B][L|C|R] (top/⏎
     …middle/bottom/left/center/right) and grid spacing must be specified via
     -I<dx>[/<dy>] (also see -r).
grdimage [ERROR]: Offending option -R///
pscoast [ERROR]: Option -J parsing failure. Correct syntax:

  -J Select map proJection (<scale> in cm/degree, <width> in cm). Modifiers:
       +dh Specifying map height.
       +du Specifying maximum map dimension.
       +dl Specifying minimum map dimension.
       +dw Specifying map width [Default].
     Azimuthal projections set -Rg unless polar aspect or -R<...>+r is set.
     -Ja|A<lon0>/<lat0>[/<hor>]/<scl> (or <radius>/<lat>)|<width> (Lambert
       Azimuthal Equal-area)
     -Jb|B<lon0>/<lat0>/<lat1>/<lat2>/<scl>|<width> (Albers Conic Equal-
       area)
     -Jcyl_stere|Cyl_stere/[<lon0>/[<lat0>/]]<lat1>/<lat2>/<scl>|<width>
       (Cylindrical Stereographic)
     -Jc|C<lon0>/<lat0><scl>|<width> (Cassini)
     -Jd|D<lon0>/<lat0>/<lat1>/<lat2>/<scl>|<width> (Equidistant Conic)
     -Je|E<lon0>/<lat0>[/<horizon>]/<scl> (or <radius>/<lat>)|<width>
       (Azimuthal Equidistant)
     -Jf|F<lon0>/<lat0>[/<horizon>]/<scl> (or <radius>/<lat>)|<width>
       (Gnomonic)
     -Jg|G<lon0>/<lat0>/<scl> (or <radius>/<lat>)|<width> (Orthographic)
     -Jg|G<lon0>/<lat0>/<scl>|<width>[+a<azimuth>][+t<tilt>][+v<vwidth>/⏎
       …<vheight>][+w<twist>][+z<altitude>[r|R]|g] (General Perspective)
     -Jh|H[<lon0>/]<scl>|<width> (Hammer-Aitoff)
     -Ji|I[<lon0>/]<scl>|<width> (Sinusoidal)
     -Jj|J[<lon0>/]<scl>|<width> (Miller)
     -Jkf|Kf[<lon0>/]<scl>|<width> (Eckert IV)
     -Jks|Ks[<lon0>/]<scl>|<width> (Eckert VI)
     -Jl|L<lon0>/<lat0>/<lat1>/<lat2>/<scl>|<width> (Lambert Conformal Conic)
     -Jm|M[<lon0>/[<lat0>/]]<scl>|<width> (Mercator)
     -Jn|N[<lon0>/]<scl>|<width> (Robinson projection)
     -Jo|O (Oblique Mercator). Specify one of three definitions:
       -Jo|O[a|A]<lon0>/<lat0>/<azimuth>/<scl>|<width>[+v]
       -Jo|O[b|B]<lon0>/<lat0>/<lon1>/<lat1>/<scl>|<width>[+v]
       -Jo|Oc|C<lon0>/<lat0>/<lonp>/<latp>/<scl>|<width>[+v]
     -Jpoly|Poly/[<lon0>/[<lat0>/]]<scl>|<width> ((American) Polyconic)
     -Jq|Q[<lon0>/[<lat0>/]]<scl>|<width> (Equidistant Cylindrical)
     -Jr|R[<lon0>/]<scl>|<width> (Winkel Tripel)
     -Js|S<lon0>/<lat0>/[<horizon>/]<scl> (or <slat>/<scl> or <radius>/<lat>)|⏎
       …<width> (Stereographic)
     -Jt|T<lon0>/[<lat0>/]<scl>|<width> (Transverse Mercator)
     -Ju|U[<zone>/]<scl>|<width> (UTM)
     -Jv|V<lon0>/<scl>|<width> (van der Grinten)
     -Jw|W<lon0>/<scl>|<width> (Mollweide)
     -Jy|Y[<lon0>/[<lat0>/]]<scl>|<width> (Cylindrical Equal-area)
     -Jp|P<scl>|<width>[+a][+f[e|p|<radius>]][+k<kind>][+r<offset>][+t<origin]⏎
       …[+z[p|<radius>]] (Polar [azimuth] (theta,radius)).
     -Jx|X<x-scl>|<width>[d|l|p<power>|t|T][/<y-scl>|<height>[d|l|p<power>|t|⏎
       …T]] (Linear, log, and power projections)
     (See basemap for more details on projection syntax)
pscoast [ERROR]: Offending option -Jmc
pscoast [ERROR]: Option -R parsing failure. Correct syntax:

  -R<west>/<east>/<south>/<north>[+r]
     Specify the min/max coordinates of your data region in user units. Use
     dd:mm[:ss] for regions given in arc degrees, minutes [and seconds]. Use
     -R<xmin>/<xmax>/<ymin>/<ymax>[+u<unit>] for regions given in projected
     coordinates, with <unit> selected from e|f|k|M|n|u [e]. If +u is set,
     projected regions centered on (0,0) may beset via -R<halfwidth>[/⏎
     …<halfheight>]+u<unit>, where <halfheight> defaults to <halfwidth> if not
     given. Use [yyyy[-mm[-dd]]]T[hh[:mm[:ss[.xxx]]]] format for time axes.
     Append +r if -R specifies the coordinates of the lower left and upper
     right corners of a rectangular area.
     Alternatively, use -R<code><x0>/<y0>/<n_columns>/<n_rows> for origin and
     grid dimensions, where <code> is a 2-char combo from [T|M|B][L|C|R] (top/⏎
     …middle/bottom/left/center/right) and grid spacing must be specified via
     -I<dx>[/<dy>] (also see -r).
pscoast [ERROR]: Offending option -R///
psscale [ERROR]: Option -J parsing failure. Correct syntax:

  -J Select map proJection (<scale> in cm/degree, <width> in cm). Modifiers:
       +dh Specifying map height.
       +du Specifying maximum map dimension.
       +dl Specifying minimum map dimension.
       +dw Specifying map width [Default].
     Azimuthal projections set -Rg unless polar aspect or -R<...>+r is set.
     -Ja|A<lon0>/<lat0>[/<hor>]/<scl> (or <radius>/<lat>)|<width> (Lambert
       Azimuthal Equal-area)
     -Jb|B<lon0>/<lat0>/<lat1>/<lat2>/<scl>|<width> (Albers Conic Equal-
       area)
     -Jcyl_stere|Cyl_stere/[<lon0>/[<lat0>/]]<lat1>/<lat2>/<scl>|<width>
       (Cylindrical Stereographic)
     -Jc|C<lon0>/<lat0><scl>|<width> (Cassini)
     -Jd|D<lon0>/<lat0>/<lat1>/<lat2>/<scl>|<width> (Equidistant Conic)
     -Je|E<lon0>/<lat0>[/<horizon>]/<scl> (or <radius>/<lat>)|<width>
       (Azimuthal Equidistant)
     -Jf|F<lon0>/<lat0>[/<horizon>]/<scl> (or <radius>/<lat>)|<width>
       (Gnomonic)
     -Jg|G<lon0>/<lat0>/<scl> (or <radius>/<lat>)|<width> (Orthographic)
     -Jg|G<lon0>/<lat0>/<scl>|<width>[+a<azimuth>][+t<tilt>][+v<vwidth>/⏎
       …<vheight>][+w<twist>][+z<altitude>[r|R]|g] (General Perspective)
     -Jh|H[<lon0>/]<scl>|<width> (Hammer-Aitoff)
     -Ji|I[<lon0>/]<scl>|<width> (Sinusoidal)
     -Jj|J[<lon0>/]<scl>|<width> (Miller)
     -Jkf|Kf[<lon0>/]<scl>|<width> (Eckert IV)
     -Jks|Ks[<lon0>/]<scl>|<width> (Eckert VI)
     -Jl|L<lon0>/<lat0>/<lat1>/<lat2>/<scl>|<width> (Lambert Conformal Conic)
     -Jm|M[<lon0>/[<lat0>/]]<scl>|<width> (Mercator)
     -Jn|N[<lon0>/]<scl>|<width> (Robinson projection)
     -Jo|O (Oblique Mercator). Specify one of three definitions:
       -Jo|O[a|A]<lon0>/<lat0>/<azimuth>/<scl>|<width>[+v]
       -Jo|O[b|B]<lon0>/<lat0>/<lon1>/<lat1>/<scl>|<width>[+v]
       -Jo|Oc|C<lon0>/<lat0>/<lonp>/<latp>/<scl>|<width>[+v]
     -Jpoly|Poly/[<lon0>/[<lat0>/]]<scl>|<width> ((American) Polyconic)
     -Jq|Q[<lon0>/[<lat0>/]]<scl>|<width> (Equidistant Cylindrical)
     -Jr|R[<lon0>/]<scl>|<width> (Winkel Tripel)
     -Js|S<lon0>/<lat0>/[<horizon>/]<scl> (or <slat>/<scl> or <radius>/<lat>)|⏎
       …<width> (Stereographic)
     -Jt|T<lon0>/[<lat0>/]<scl>|<width> (Transverse Mercator)
     -Ju|U[<zone>/]<scl>|<width> (UTM)
     -Jv|V<lon0>/<scl>|<width> (van der Grinten)
     -Jw|W<lon0>/<scl>|<width> (Mollweide)
     -Jy|Y[<lon0>/[<lat0>/]]<scl>|<width> (Cylindrical Equal-area)
     -Jp|P<scl>|<width>[+a][+f[e|p|<radius>]][+k<kind>][+r<offset>][+t<origin]⏎
       …[+z[p|<radius>]] (Polar [azimuth] (theta,radius)).
     -Jx|X<x-scl>|<width>[d|l|p<power>|t|T][/<y-scl>|<height>[d|l|p<power>|t|⏎
       …T]] (Linear, log, and power projections)
     (See basemap for more details on projection syntax)
psscale [ERROR]: Offending option -Jmc
psscale [WARNING]: No country code matching azi_offset_ll.grd (skipped)
psconvert [ERROR]: The file azioff_ll.ps has no BoundingBox in the first 20 lines or last 256 bytes. Use -A option.
Azimuth/Offset map: azioff_ll.png
root@ubuntu:~/M-SAR/F1#

**Comment by Xiaohua-Eric-Xu** [comment, marked as answer]
`make_a_offset.csh S1_20220105_111101_F1.PRM S1_20220117_111101_F1.PRM 8 2 8 2 1`
You may have set nx and ny too small, also x_search and y_search too small as well. Use larger numbers such as `500 500 128 128 1`. It'll take quite a while to run though.

**Reply by KIKINEO** [reply]
Dear Mr. Xu, Hello, I have a question that I would like to sincerely invite you to answer, which is about the incident angle and azimuth angle of Sentinel-1 data. I have some confusion, that is, the parameters of the incident angle and azimuth angle of each image. Is it the same? I now want to use these parameters to solve the three components of the los-direction deformation variable. Generally, this work is to treat the data of the same track or image as the same? Or is the incident angle and reflection angle of each pixel different? I sincerely look forward to your answer, best wishes~ @Xiaohua-Eric-Xu

**Reply by Xiaohua-Eric-Xu** [reply]
The azimuth should be about the same unless you have a very long coverage along track. The incidence angle however, changes with range. You could get the look vector for any given point using `SAT_look`.

**Reply by KIKINEO** [reply]
I am grateful for your help and advice and I will continue to study it~
